### PR TITLE
Conform more of the codebase to `strictNullChecks` (#10358

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/element-web/issues/21967

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/elements/EffectsOverlay.tsx [hunk 1]
```diff
@@ -30,15 +30,15 @@ const EffectsOverlay: FunctionComponent<IProps> = ({ roomWidth }) => {
...
+    const lazyLoadEffectModule = async (name: string): Promise<ICanvasEffect | null> => {
         if (!name) return null;
         let effect: ICanvasEffect | null = effectsRef.current.get(name) || null;
         if (effect === null) {
             const options = CHAT_EFFECTS.find((e) => e.command === name)?.options;
             try {
                 const { default: Effect } = await import(`../../../effects/${name}`);
                 effect = new Effect(options);
-                effectsRef.current.set(name, effect);
+                effectsRef.current.set(name, effect!);
```
Comment (weeman1337) [inline src/components/views/elements/EffectsOverlay.tsx:41]: `effect` is created one line before. Does typescript really not know that `effect` exists here?

File: src/components/views/elements/DialogButtons.tsx [hunk 2]
```diff
@@ -69,17 +69,17 @@ export default class DialogButtons extends React.Component<IProps> {
     };
 
     private onCancelClick = (event: React.MouseEvent): void => {
-        this.props.onCancel(event);
+        this.props.onCancel!(event);
```
Comment (weeman1337) [inline src/components/views/elements/DialogButtons.tsx:72]: In my opinion there should be a test that the `onCancel` is set [here](https://github.com/matrix-org/matrix-react-sdk/pull/10358/files#diff-46b24f2bf826b8f8dc45f660caef76a06fec748dd25a30b00c5d73cf6733c076R82).

Or this should better be 
```suggestion
        this.props.onCancel?(event);
```

File: src/components/views/messages/EditHistoryMessage.tsx [hunk 3]
```diff
@@ -59,11 +59,11 @@ export default class EditHistoryMessage extends React.PureComponent<IProps, ISta
         super(props);
 
         const cli = MatrixClientPeg.get();
-        const { userId } = cli.credentials;
+        const userId = cli.getSafeUserId();
         const event = this.props.mxEvent;
         const room = cli.getRoom(event.getRoomId());
         if (event.localRedactionEvent()) {
-            event.localRedactionEvent().on(MatrixEventEvent.Status, this.onAssociatedStatusChanged);
+            event.localRedactionEvent()!.on(MatrixEventEvent.Status, this.onAssociatedStatusChanged);
```
Comment (weeman1337) [inline src/components/views/messages/EditHistoryMessage.tsx:66]: The `if` condition can be replaced by

```suggestion
            event.localRedactionEvent()?.on(MatrixEventEvent.Status, this.onAssociatedStatusChanged);
```

(same for other if below)

File: src/components/views/rooms/E2EIcon.tsx [hunk 4]
```diff
@@ -78,9 +78,9 @@ const E2EIcon: React.FC<IProps> = ({
 
     let e2eTitle: string | undefined;
     if (isUser) {
-        e2eTitle = crossSigningUserTitles[status];
+        e2eTitle = crossSigningUserTitles[status!];
```
Comment (weeman1337) [inline src/components/views/rooms/E2EIcon.tsx:81]: If assuming, that the status is always set,
why changing the prop interface to make it optional?

File: src/components/views/elements/EffectsOverlay.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/views/elements/EffectsOverlay.tsx:41]: Effect is untyped due to `await import(`../../../effects/${name}`);`

File: src/components/views/rooms/E2EIcon.tsx (hunk 4 above)
Comment (t3chguy) [reply src/components/views/rooms/E2EIcon.tsx:81]: `{}[undefined] === undefined` so it was just a trick to skip needing another if case for if `status === undefined`

**Pull request merged**